### ISSUE TITLE
startCopyBlob error: a mandatory request header is missing

### ISSUE THREAD

**Issue by ritazh** [Azure/azure-storage-node#71]
I think we need to add:
```
"x-ms-blob-type": "BlockBlob"
```

**Comment by yaxia** [comment]
@ritazh Thanks for your interests in the library. From https://msdn.microsoft.com/en-us/library/azure/dd894037.aspx, we don't support blob type in the request headers. Would you please share what your scenario is that you need it? Thanks.

**Comment by ritazh** [comment]
Thanks @yaxia I am trying to use this SDK to copy blobs from one container to another. In my code, I created a blob service, then called startCopyBlob with the blob Uri, the target container name, and the target blob name. 
```
service.startCopyBlob(sourceUri, targetContainerName, targetBlobName ...
```
The returned response error was "a mandatory request header is missing. specifically "x-ms-blob-type". I added that header in the request and everything worked. Please see this PR: https://github.com/Azure/azure-storage-node/pull/72

**Comment by yaxia** [comment]
@ritazh The server's behavior is not expected. I cannot reproduce the issue at my side. Appreciate if you don't mind enable the verbose log and share the outputs. (Note: Storage account name will be printed and no other private information will be exposed.)
```
var storage = require('azure-storage');
service.logger.level = storage.Logger.LogLevels.DEBUG;
service.startCopyBlob(sourceUri, targetContainerName, targetBlobName ...
```

**Comment by yaxia** [comment]
When calling the blobService. startCopyBlob() and missing the source URL (the first parameter: sourceUri ), there will be no “x-ms-copy-source” header in the request, so the server will take it as a [Put Blob](https://msdn.microsoft.com/en-us/library/azure/dd179451.aspx) call instead of a [Copy Blob](https://msdn.microsoft.com/en-us/library/azure/dd894037.aspx) call. Ensuring the sourceUri valid in service.startCopyBlob(sourceUri, targetContainerName, targetBlobName) could mitigate the issue.
 
Anyway, it is an issue that the SDK doesn’t throw error when the sourceUri is invalid. Thanks for reporting the issue. We will fix it in a later release.

**Comment by yaxia** [comment]
Fixed in 0.6.0. Thanks.
